Implement Vibes API

hawkinse opened the pull request.

ginge commented on rwatch/ui/vibes.h:
```diff
@@ -0,0 +1,20 @@
+
+#include <stdint.h>
+
+typedef struct VibePattern{
```
space on brace please

ginge commented on rcore/main.c:
```diff
@@ -16,11 +16,11 @@ int main(void)
 
     rebbleos_init();
-    
+
```
not sure about touching this file as part of the PR. But I do agree it is messy

ginge commented:
Looks great, thanks!
A couple of really petty minors in there, but happy to merge.
However, vibrate currently doesn't work on hardware. I think there are some build errors and certainly some null deref somewhere. I would like to fix that before we turn vibe back on.

hawkinse commented:
I've added the brace spaces and removed changes to main.c. At this point it'll lock up on use due to the commented out vibrate_init(), merging should probably wait until this is addressed.

ginge commented:
superb work. lovely test too. thanks!
